subsys/gazell/gzp_device.c

```diff
@@ -530,13 +540,14 @@ void gzp_init(void)
 	(void)gzp_update_radio_params(gzp_system_address);
 }
 
-
 void gzp_erase_pairing_data(void)
 {
 #ifdef CONFIG_GAZELL_PAIRING_SETTINGS
 	settings_delete(db_key_sys_addr);
+	gzp_clear_sys_addr();
 #ifdef CONFIG_GAZELL_PAIRING_CRYPT
 	settings_delete(db_key_host_id);
+	gzp_clear_host_id();
```

Comment:
Apart from clearing RAM, shouldn't we update the radio parameters as we do in `gzp_init`? Or it's not necessary here?
```
        /* Update radio parameters from gzp_system_address */                                        
        (void)gzp_update_radio_params(gzp_system_address);
```

Reply:
In nRF5 SDK, this function erases the flash page. Therefore, reading of the flash variables gives 0xFF. NCS forget clearing the RAM variables. This fix gets us aligned with the nRF5 SDK.

To keep things as is, I hesitate to update the radio parameters here. When Gazell is running, the user should not call this function to erase the system address. The user is expected to know what they are doing... It gets tricky sometimes.

Reply:
I would not insist on `gzp_update_radio_params` (I just noticed it in `gzp_init` and therefore asked).

btw. I think we should clear RAM data also if `CONFIG_GAZELL_PAIRING_SETTINGS` is disabled (overlooked it previously).

Reply:
Clear system address and host_id whether `CONFIG_GAZELL_PAIRING_SETTINGS` is enabled or not.

Reply:
The gzp_host_id() call will not build if GAZELL_PAIRING_CRYPT option is disabled.

Reply:
@pirun I have pushed a commit fixing the gzp_host_id() with CONFIG_GAZELL_PAIRING_CRYPT. Please check it out and merge it to your development branch.